### PR TITLE
Add oui-rule selector to horizontal rule css #1016

### PR DESCRIPTION
**Pull request by jessicalc**
This adds a `.oui-rule` selector in so we don't need to keep referring to lego classes!! 

See #1016

### CODE REVIEW

File: src/oui/partials/elements/_horizontal-rules.scss [hunk 1]
```diff
@@ -11,7 +11,8 @@
 ////
 
 %#{$namespace}rule,
-.#{$namespace}rule {
+.#{$namespace}rule,
```
Comment (circAssimilate) [inline src/oui/partials/elements/_horizontal-rules.scss]: Shouldn't `namespace` pick this up - or am I missing what that does?

**Comment by alansun-optimizely** [comment]
Could you also add a HorizontalRule react component?

File: src/oui/partials/elements/_horizontal-rules.scss (hunk 1 above)
Comment (jessicalc) [reply src/oui/partials/elements/_horizontal-rules.scss]: It doesn't pick it up; anything without `oui-` as a prefix won't be available in `oui`. I'm not sure the historic reason for this, but I think @daverau-optimizely will know?

**Comment by jessicalc** [comment]
> Could you also add a HorizontalRule react component?

Not within scope of ticket, but we should ticket it up in JIRA.

File: src/oui/partials/elements/_horizontal-rules.scss (hunk 1 above)
Comment (jessicalc) [reply src/oui/partials/elements/_horizontal-rules.scss]: https://github.com/optimizely/oui/blob/5d7bd8b67a6c0fdfbf4ef606dd406f0e940a142d/src/oui/_oui-variables.scss#L12

**Pull request merged**